### PR TITLE
Keeper tx gasprice feature

### PR DESCRIPTION
**Pull request by RyanRHall**

### CODE REVIEW

File: core/config/general_config.go [hunk 1]
```diff
@@ -782,6 +783,11 @@ func (c *generalConfig) KeeperRegistrySyncUpkeepQueueSize() uint32 {
 	return c.getWithFallback("KeeperRegistrySyncUpkeepQueueSize", ParseUint32).(uint32)
 }
 
+// KeeperCheckUpkeepGasPriceFeatureEnabled enables keepers to include a gas price when running checkUpkeep
+func (c *generalConfig) KeeperCheckUpkeepGasPriceFeatureEnabled() bool {
+	return c.viper.GetBool(EnvVarName("KeeperCheckUpkeepGasPriceFeatureEnabled"))
```
Comment (samsondav) [inline core/config/general_config.go:788]: Does this fall back to the default? I think you may want `c.getWithFallback`

File: core/services/keeper/delegate.go [hunk 2]
```diff
@@ -104,6 +104,7 @@ func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.Service, err er
 		chain.TxManager().GetGasEstimator(),
 		svcLogger,
 		chain.Config(),
+		chain.Config().KeeperCheckUpkeepGasPriceFeatureEnabled(),
```
Comment (samsondav) [inline core/services/keeper/delegate.go:107]: This seems unnecessary - it's already on the config, why not simply read it from there?

File: core/services/pipeline/variables.go [hunk 3]
```diff
@@ -85,10 +85,28 @@ func (vars Vars) Set(dotID string, value interface{}) {
...
+		case map[string]interface{}:
+			parent = child
+		case nil:
+			newChild := make(map[string]interface{})
+			parent[key] = newChild
+			parent = newChild
+		default:
+			panic("cannot set a key of a non map[string]interface{} type")
+		}
+	}
```
Comment (samsondav) [inline core/services/pipeline/variables.go:109]: This change seems unrelated?

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/services/keeper/delegate.go (hunk 2 above)
Comment (RyanRHall) [reply core/services/keeper/delegate.go:107]: ah you're right

File: core/config/general_config.go (hunk 1 above)
Comment (RyanRHall) [reply core/config/general_config.go:788]: also right :+1:

File: core/services/pipeline/variables.go (hunk 3 above)
Comment (RyanRHall) [reply core/services/pipeline/variables.go:109]: it was necessary in order to do `vars.Set("jobSpec.gasPrice", gasPrice)` etc in `upkeep_executer.go`

Comment (RyanRHall) [reply core/services/pipeline/variables.go:109]: possible to do without, but this was cleaner

File: core/services/pipeline/variables.go [hunk 4]
```diff
@@ -85,10 +85,28 @@ func (vars Vars) Set(dotID string, value interface{}) {
...
+		}
+		switch child := parent[key].(type) {
+		case map[string]interface{}:
+			parent = child
+		case nil:
+			newChild := make(map[string]interface{})
+			parent[key] = newChild
+			parent = newChild
+		default:
+			panic("cannot set a key of a non map[string]interface{} type")
```
Comment (begmaroman) [inline core/services/pipeline/variables.go:107]: Maybe just throw an error instead of panicking in runtime?

**Review by begmaroman — APPROVED** [review]
LGTM, just left one comment and agree with Sam's review

File: core/services/pipeline/variables.go (hunk 4 above)
Comment (RyanRHall) [reply core/services/pipeline/variables.go:107]: wanted to keep it consistent with previous functionality, which panics

File: core/services/pipeline/variables.go (hunk 3 above)
Comment (samsondav) [reply core/services/pipeline/variables.go:109]: This block of code makes me nervous.

Why not instead set a nil gasFee, gasTip etc and fil those vars in if the feature flag is enabled?

File: core/services/keeper/upkeep_executer.go [hunk 5]
```diff
@@ -196,13 +190,21 @@ func (ex *UpkeepExecuter) execute(upkeep UpkeepRegistration, headNumber int64, d
 			"performUpkeepGasLimit": upkeep.ExecuteGas + ex.orm.config.KeeperRegistryPerformGasOverhead(),
 			"checkUpkeepGasLimit": ex.config.KeeperRegistryCheckGasOverhead() + uint64(upkeep.Registry.CheckGas) +
 				ex.config.KeeperRegistryPerformGasOverhead() + upkeep.ExecuteGas,
-			"gasPrice":   gasPrice,
-			"gasTipCap":  fee.TipCap,
-			"gasFeeCap":  fee.FeeCap,
```
Comment (samsondav) [inline core/services/keeper/upkeep_executer.go:201]: I would recommend keeping these here and moving the estimateGas call up. Like so:

```go
var gasPrice, gasTipCap, gasFeeCap *big.Int

if ex.config.KeeperCheckUpkeepGasPriceFeatureEnabled() {
  gasPrice = foo
   ...
   ```

**Review by samsondav — CHANGES_REQUESTED** [review]

**Review by jmank88 — APPROVED** [review]
